src/main/java/org/opensearch/security/configuration/ConfigurationLoaderSecurity7.java

```diff
@@ -148,8 +148,8 @@ public void noData(String id) {
 
                 // Since NODESDN is newly introduced data-type applying for existing clusters as well, we make it backward compatible by returning valid empty
                 // SecurityDynamicConfiguration.
-                // Same idea for new setting WHITELIST
-                if (cType == CType.NODESDN || cType == CType.WHITELIST) {
+                // Same idea for new setting WHITELIST/ALLOWLIST
+                if (cType == CType.NODESDN || cType == CType.WHITELIST || cType == CType.ALLOWLIST) {
```

Comment:
Good catch! add deprecation on whitelist API path.

Reply:
This is [a known issue in OpenSearch](https://github.com/opensearch-project/OpenSearch/issues/3290). This is the approach to deprecate APIs since ElasticSearch. I don't think this is a worthwhile blocker on this PR for 2.0.0.

Reply:
Thanks for filing the issue, resolving this thread.